server/socket.ts

```diff
@@ -25,39 +37,52 @@ export function listenSocket(server) {
     socket.on("disconnect", async () => {
```

Comment:
As this file potentially grows, think about extracting your eventhandlers from setting up your server. There could be LobbyEvents, GameEvents and so on. These could then be added to a `regsiteredEvents` array. How about ending up with something like:

```
// Dont use `Function` as a type. It is late and I am lazy :P
registeredEvents.forEach( (event: { name: string, handler: Function}) => {
  const {name, handler} = event;
  socket.on(name, event);
})
``` 

Keeping structure to your events will help you significally in the long run as it will be harder and harder to manage them the more there are.

Reply:
This sounds very interesting - even though right now I don't see how this works, or rather where I would implement these event arrays 🤔